cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -257,33 +286,57 @@ func (p *dockerProject) Package(
 ) *async.TaskWithProgress[*ServicePackageResult, ServiceProgress] {
 	return async.RunTaskWithProgress(
 		func(task *async.TaskContextWithProgress[*ServicePackageResult, ServiceProgress]) {
-			imageId := buildOutput.BuildOutputPath
+			var imageId string
+
+			if buildOutput != nil {
+				imageId = buildOutput.BuildOutputPath
+			}
+
+			packageDetails := &dockerPackageResult{
+				ImageHash: imageId,
+			}
+
+			// If we don't have an image ID from a docker build then an external source image is being used
 			if imageId == "" {
-				task.SetError(errors.New("missing container image id from build output"))
-				return
+				sourceImage, err := docker.ParseContainerImage(serviceConfig.Image)
+				if err != nil {
+					task.SetError(fmt.Errorf("parsing source container image: %w", err))
+					return
+				}
+
+				remoteImageUrl := sourceImage.Remote()
+
+				task.SetProgress(NewServiceProgress("Pulling container source image"))
+				if err := p.docker.Pull(ctx, remoteImageUrl); err != nil {
+					task.SetError(fmt.Errorf("pulling source container image: %w", err))
+					return
+				}
+
+				imageId = remoteImageUrl
+				packageDetails.SourceImage = remoteImageUrl
 			}
 
-			localTag, err := p.containerHelper.LocalImageTag(ctx, serviceConfig)
+			// Generate a local tag from the 'docker' configuration section of the service
+			imageWithTag, err := p.containerHelper.LocalImageTag(ctx, serviceConfig)
```

Comment:
Is `azd package <image service>` pulling down the image repository and tagging it the expected behavior users want? It certainly feels natural in relation to the other features. It could also seem out-of-place since the thing we're "packaging" is already packaged.

Reply:
You cannot create a new tag on an existing image without first having the image locally.  If you're working with large external images then it will slow down your package/deploy call but I don't see anyway around it at the moment.

So if I wanted to tag/rename a public image "nginx" I would first need to pull it locally, then tag & push it.
If someone were to attempt to replicate this behavior in a script/hook they would need to do the following:

```bash
docker pull nginx
docker tag nginx contoso.azurecr.io/nginx:my-custom-tag
docker push contoso.azurecr.io/nginx:my-custom-tag
```